library/waku_thread/config.nim

```diff
@@ -191,76 +177,41 @@ proc parseConfig*(
   var listenAddr: IpAddress
   try:
     listenAddr = parseIpAddress("127.0.0.1")
-    if not parseListenAddr(jsonNode, listenAddr, errorResp):
+    if not parseListenAddr(jsonNode, conf, errorResp):
       return false
   except Exception, ValueError:
     errorResp = "Exception calling parseIpAddress: " & getCurrentExceptionMsg()
     return false
 
   # port
-  var port = 0
   try:
-    if not parsePort(jsonNode, port, errorResp):
+    if not parsePort(jsonNode, conf, errorResp):
       return false
   except Exception, ValueError:
     errorResp = "Exception calling parsePort: " & getCurrentExceptionMsg()
     return false
 
-  let natRes = setupNat("any", clientId, Port(uint16(port)), Port(uint16(port)))
-  if natRes.isErr():
-    errorResp = "failed to setup NAT: " & $natRes.error
-    return false
-
-  let (extIp, extTcpPort, _) = natRes.get()
-
-  let extPort =
-    if extIp.isSome() and extTcpPort.isNone():
-      some(Port(uint16(port)))
-    else:
-      extTcpPort
-
```

Comment:
We don't need this because the conf code does it already right?

edit: nvm the answer is yes https://github.com/waku-org/nwaku/blob/18a053591d8d18b8f066d4bc919c3910cb4295c3/waku/factory/internal_config.nim#L80